models/Asset/Folder.php

```diff
@@ -89,15 +76,7 @@ public function getChildren(): array
 
     public function hasChildren(): bool
     {
-        if (is_bool($this->hasChildren)) {
-            if (($this->hasChildren && empty($this->children)) || (!$this->hasChildren && !empty($this->children))) {
-                return $this->getDao()->hasChildren();
-            }
-
-            return $this->hasChildren;
-        }
-
-        return $this->getDao()->hasChildren();
+      return $this->getDao()->hasChildren();
```

Comment:
It is not worth it to cache this data as anyway you have to check against if the children exist on db. see example below:

```php
$parent = new Asset\Folder();
$child = new Asset();
$child->setParent($parent);
$child->save();

$parent->hasChildren(); // should return true
$child->delete();

$parent->hasChildren(); //should return false
```
if this change is approved, then we should consider this for `AbstractObject` & `Document`.

Reply:
Same scenario for `hasSibling` property.

Reply:
Changed for `AbstractObject` & `Document` along with `hasSibling` changes.